### PR TITLE
Review and fix TODO code snippet references (docs 2)

### PR DESCRIPTION
**Pull request by tompratt-AQ**
# Title
Review and fix all TODO code snippet references.  (\docs Part 2)

## Summary
This PR covers the rest of the conceptual files (\docs) that are listed in the dotnet/docs Project "TODO: review snippet reference". It also fixes issues with .config files from the previous #1982 by calling them with an XML code tag. 

## Details
Where there are two identical config files, the one in the language folder (and common subfolder) is used instead of the common folder.

## Suggested Reviewers
@mairaw

### CODE REVIEW

**Comment by tompratt-AQ** [comment]
The following comments are for the changes in this PR:
1. The file docs/framework/wcf/feature-details/how-to-host-a-wcf-service-in-iis.md has an inline xml snippet that is the same as the xml .config snippet file that is being called. Should the inline code be replaced?
2. I added code identifiers to several inline code snippets in some of the files. 
3. I changed snippet tags from csharp style comments to XML style comments in some config files. Since we are now calling them with an XML code tag, I was worried it might not find them.

**Comment by rpetrusha** [comment]
@tompratt-AQ, to respond to your question in #1, if there's a choice between incline code and external code, it's best to use external code. (It's more maintainable.)

File: samples/snippets/csharp/VS_Snippets_CFX/c_standalonedatacontractserializer/cs/source.cs [hunk 1]
```diff
@@ -42,19 +42,19 @@ public class LibraryPatron
         [DataContract]
         public class LibraryItem
         {
-            //code not shown�
+            //code not shown�
```
Comment (rpetrusha) [inline samples/snippets/csharp/VS_Snippets_CFX/c_standalonedatacontractserializer/cs/source.cs]: What is the final character (rendered as a question mark in a triangle) supposed to be? Make sure the file is saved using UTF-8 encoding (it's probably saved using ASCII encoding).

**Review by rpetrusha — APPROVED** [review]
LGTM, except for one comment about the final character in one C# source code file. And, in response to my last comment, replacing inline code with an external code reference.

**Comment by tompratt-AQ** [comment]
I addressed Ron's comments, fixed 4 more files, and fixed one file by re-adding snippets. 

Once these clear, there is only one file left in \docs with "<!-- TODO: review snippet reference" in it:
docs/standard/base-types/how-to-display-localized-date-and-time-information-to-web-users.md which has a snippet tag that links to an .aspx file. Before filing a bug on the issue, it seems like there should be a way to make it work with our current snippet tag system.

**Comment by tompratt-AQ** [comment]
Filed issue [982841](https://mseng.visualstudio.com/VSChina/_workitems?id=982841) about the code snippet that links to the .aspx files. So, we'll leave that TODO as the last one in docs. If everything else  checks out here, this should be ready to go!

File: docs/framework/wcf/creating-ws-i-basic-profile-1-1-interoperable-services.md [hunk 2]
```diff
@@ -41,8 +41,8 @@ To configure a [!INCLUDE[indigo2](../../../includes/indigo2-md.md)] service endp
   
 ### Code  
  [!code-csharp[C_HowTo-WCFServiceAndASMXClient#0](../../../samples/snippets/csharp/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/cs/program.cs#0)]
- [!code-vb[C_HowTo-WCFServiceAndASMXClient#0](../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/vb/program.vb#0)]  
- [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../samples/snippets/common/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)]
+ [!code-vb[C_HowTo-WCFServiceAndASMXClient#0](../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/vb/program.vb#0)] 
+ [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../samples/snippets/csharp/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)]
```
Comment (mairaw) [inline docs/framework/wcf/creating-ws-i-basic-profile-1-1-interoperable-services.md]: I'd say as we start changing the references from common to csharp that we also pull the snippet out at the same time and also make sure no other topic is referencing that file.

File: docs/framework/wcf/feature-details/config-wcf-service-with-aspnet-web-service.md [hunk 3]
```diff
@@ -47,7 +47,8 @@ To configure a [!INCLUDE[indigo1](../../../../includes/indigo1-md.md)] service e
   
  [!code-csharp[C_HowTo-WCFServiceAndASMXClient#0](../../../../samples/snippets/csharp/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/cs/program.cs#0)] 
  [!code-vb[C_HowTo-WCFServiceAndASMXClient#0](../../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/vb/program.vb#0)] 
- <!-- TODO: review snippet reference [!code[C_HowTo-WCFServiceAndASMXClient#1](../../../../samples/snippets/common/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)] -->    
+ [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../../samples/snippets/csharp/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)] 
+ [!code-xml[C_HowTo-WCFServiceAndASMXClient#1](../../../../samples/snippets/visualbasic/VS_Snippets_CFX/c_howto-wcfserviceandasmxclient/common/app.config#1)]
```
Comment (mairaw) [inline docs/framework/wcf/feature-details/config-wcf-service-with-aspnet-web-service.md]: remove this duplicate snippet.

File: samples/snippets/common/VS_Snippets_CFX/c_howto_configureclientbinding/common/client.exe.config [hunk 4]
```diff
@@ -1,5 +1,5 @@
-//<snippet100>
-<?xml version="1.0" encoding="utf-8" ?>
+<?xml version="1.0" encoding="utf-8" ?>   
+<!-- <snippet100> -->
```
Comment (mairaw) [inline samples/snippets/common/VS_Snippets_CFX/c_howto_configureclientbinding/common/client.exe.config]: shouldn't this come before as in the original?

File: samples/snippets/common/VS_Snippets_CFX/c_howto_createreliablesessionhttps/common/app.config [hunk 5]
```diff
@@ -1,7 +1,7 @@
-//This is the config for the topic: How to: Create a Custom Reliable Session Binding with HTTPS
-//<snippet2211>
+<!-- This is the config for the topic: How to: Create a Custom Reliable Session Binding with HTTPS -->
 
 <?xml version="1.0" encoding="utf-8" ?>
+<!-- <snippet2211> -->
```
Comment (mairaw) [inline samples/snippets/common/VS_Snippets_CFX/c_howto_createreliablesessionhttps/common/app.config]: same here and in some of the samples below

**Review by mairaw — CHANGES_REQUESTED** [review]
This is great @tompratt-AQ! Left a few comments.

**Comment by tompratt-AQ** [comment]
@mairaw In this last commit, I did the following:
1. Fixed the specific issues that you flagged in your last comments.
2. For all config files that were in TODO links in this PR: I confirmed that there were duplicates between the Common branch and the csharp branch, made sure that any other reference to the Common file were changed, and then deleted the config file in the Common branch.
3. Deleted other config files that were in the same folder as the deleted dupes after confirming that they also were dupes and weren't being referenced.

**Review by mairaw — APPROVED** [review]
It's great to get these fixes in. Thanks @tompratt-AQ!

**Pull request merged**